ext/XS-Typemap/Typemap.xs

```diff
@@ -917,7 +917,7 @@ T_ARRAY( dummy, array, ... )
  PREINIT:
   U32 size_RETVAL;
  CODE:
-  dummy += 0; /* Fix -Wall */
+  dummy = dummy + 0; /* GH 21505 */
```

Comment:
Don't we have `PERL_UNUSED_VAR(dummy)` for situations like this?

Reply:
I will try that out.

Reply:
> Don't we have `PERL_UNUSED_VAR(dummy)` for situations like this?

Revised; force-pushed; now smoking.  Please re-review.  Thanks.